controller/work_item_link_type_combination.go

```diff
@@ -0,0 +1,187 @@
+			}
+			return ctx.OK(res)
+		})
+	})
+}
+
+// ConvertWorkItemLinkTypeCombinationsFromModel converts between internal and external REST representation
+func ConvertWorkItemLinkTypeCombinationsFromModel(appl application.Application, request *goa.RequestData, combis []link.WorkItemLinkTypeCombination, additional ...WorkItemLinkTypeCombinationConvertFunc) ([]*app.WorkItemLinkTypeCombinationData, error) {
+	var res = []*app.WorkItemLinkTypeCombinationData{}
+	for _, i := range combis {
+		converted, err := ConvertWorkItemLinkTypeCombinationFromModel(appl, request, i, additional...)
+		if err != nil {
+			return nil, errs.WithStack(err)
+		}
+		res = append(res, converted)
+	}
+	return res, nil
+}
+
+// WorkItemLinkTypeCombinationConvertFunc is a open ended function to add additional links/data/relations to a type combination during
+// convertion from internal to API
+type WorkItemLinkTypeCombinationConvertFunc func(application.Application, *goa.RequestData, *link.WorkItemLinkTypeCombination, *app.WorkItemLinkTypeCombinationData) error
+
+func ConvertWorkItemLinkTypeCominationToModel(a app.WorkItemLinkTypeCombinationData) (link.WorkItemLinkTypeCombination, error) {
+	m := link.WorkItemLinkTypeCombination{}
+	if err := a.Validate(); err != nil {
+		return m, errs.Wrap(err, "failed to validate work item link type combination")
+	}
+	if a.ID != nil {
+		m.ID = *a.ID
+	}
+	if a.Attributes != nil {
+		if a.Attributes.Version != nil {
+			m.Version = *a.Attributes.Version
+		}
+		if a.Attributes.CreatedAt != nil {
+			m.CreatedAt = *a.Attributes.CreatedAt
+		}
+		if a.Attributes.UpdatedAt != nil {
+			m.UpdatedAt = *a.Attributes.UpdatedAt
+		}
+	}
+	rel := a.Relationships
+	if rel != nil {
+		if rel.Space != nil && rel.Space.Data != nil && rel.Space.Data.ID != nil {
+			m.SpaceID = *rel.Space.Data.ID
+		}
+		if rel.SourceType != nil && rel.SourceType.Data != nil {
+			m.SourceTypeID = rel.SourceType.Data.ID
+		}
+		if rel.TargetType != nil && rel.TargetType.Data != nil {
+			m.TargetTypeID = rel.TargetType.Data.ID
+		}
+		if rel.LinkType != nil && rel.LinkType.Data != nil {
+			m.LinkTypeID = rel.LinkType.Data.ID
+		}
+	}
+	return m, nil
+}
+
+// ConvertWorkItemLinkTypeCombinationFromModel converts a type combination from a
+// model representation to the app representation.
+func ConvertWorkItemLinkTypeCombinationFromModel(appl application.Application, request *goa.RequestData, m link.WorkItemLinkTypeCombination, additional ...WorkItemLinkTypeCombinationConvertFunc) (*app.WorkItemLinkTypeCombinationData, error) {
+	spaceSelfURL := rest.AbsoluteURL(request, app.SpaceHref(m.SpaceID.String()))
+	witTargetSelfURL := rest.AbsoluteURL(request, app.WorkitemtypeHref(m.SpaceID.String(), m.SourceTypeID.String()))
+	witSourceSelfURL := rest.AbsoluteURL(request, app.WorkitemtypeHref(m.SpaceID.String(), m.SourceTypeID.String()))
+	parentSelfURL := rest.AbsoluteURL(request, app.WorkItemLinkTypeCombinationHref(m.SpaceID.String(), m.SourceTypeID.String()))
```

Comment:
Again, good catch. I must have done too much copy'n'paste on my first day of PTO when this was written ;) Fixed in bf9aa6a3c7a827017a0220c7d9d816796df24212.